### PR TITLE
Penlight-Fixes

### PR DESCRIPTION
**Pull request by SleepyScarecrow**
# Description

Tweaked penlights some.  You can no longer examine random objects, you cannot examine your own eyes, and the penlight now needs to be actually on in order to use. Also tweaked the messages to be more accurate.

# Changelog

:cl: Tilkku
- tweak: Penlight Exam Messages
- fix: Penlights now need to be on in order to examine
- fix: Penlights can no longer be used on self

### CODE REVIEW

**Comment by SleepyScarecrow** [comment]
https://github.com/user-attachments/assets/3254d534-6117-4122-b397-aee8109ea779

**Comment by SleepyScarecrow** [comment]
why did that check fail, i cannot tell if that's my fault lol

File: Content.Server/Medical/PenLightSystem.cs [hunk 1]
```diff
@@ -22,28 +24,31 @@ public sealed class PenLightSystem : EntitySystem
...
 
-    private void OnAfterInteract(EntityUid uid, PenLightComponent component, AfterInteractEvent args)
+    private void OnAfterInteract(EntityUid uid, PenLightComponent component, ref AfterInteractEvent args)
     {
-        if (args.Handled 
-            || args.Target is not { } target)
+        if (args.Handled
+            || args.Target is not {} target)
+            return;
+        if (target == null || !args.CanReach || !HasComp<MobStateComponent>(target) || !_powerCell.HasDrawCharge(uid, user: args.User))
```
Comment (VMSolidus) [inline Content.Server/Medical/PenLightSystem.cs:42]:
```suggestion
        if (args.Handled
            || args.Target is not {} target
            || target == null 
            || !args.CanReach 
            || !HasComp<MobStateComponent>(target) 
            || !_powerCell.HasDrawCharge(uid, user: args.User))
```

File: Content.Server/Medical/PenLightSystem.cs [hunk 2]
```diff
@@ -88,6 +112,10 @@ private void Diagnose(EntityUid penlight, EntityUid target)
         if (!_uiSystem.TryGetUi(penlight, PenLightUiKey.Key, out var ui)
             || !HasComp<EyeComponent>(target))
             return;
+
+        if (!HasComp<DamageableComponent>(target))
+            return;
```
Comment (VMSolidus) [inline Content.Server/Medical/PenLightSystem.cs:117]:
```suggestion
        if (!_uiSystem.TryGetUi(penlight, PenLightUiKey.Key, out var ui)
            || !HasComp<EyeComponent>(target)
            || !HasComp<DamageableComponent>(target))
            return;
```

**Review by VMSolidus — CHANGES_REQUESTED** [review]

**Review by VMSolidus — APPROVED** [review]

**Pull request merged**